app/models/miq_cockpit_ws_worker.rb

```diff
@@ -0,0 +1,92 @@
+require 'miq_apache'
+require 'runcmd'
+require 'linux_admin'
+
+class MiqCockpitWsWorker < MiqWorker
+  require_nested :Runner
+
+  APACHE_CONF_FILE = '/etc/httpd/conf.d/manageiq-cockpit.conf'.freeze
+  class_attribute(:web_service_uri)
+
+  self.web_service_uri = "http://localhost:4000"
+  self.required_roles = ['cockpit_ws']
+  self.maximum_workers_count = 1
+
+  def friendly_name
+    @friendly_name ||= "Cockpit Worker"
+  end
+
+  def self.can_start_cockpit_ws
+    if @supports_cockpit_ws.nil?
+      @supports_cockpit_ws = MiqEnvironment::Command.is_linux? && \
+                             File.exist?('/usr/libexec/cockpit-ws') && \
+                             File.exist?('/usr/libexec/cockpit-ssh')
+    end
+    @supports_cockpit_ws
+  end
+
+  def self.can_activate_cockpit?(opts = nil)
+    return false unless has_required_role? && MiqServer.my_server.has_active_role?('web_services')
+
+    opts ||= MiqCockpitWsWorker.worker_settings
+    if opts[:external_url]
+      return true
+    end
+    can_start_cockpit_ws
+  end
+
+  # Update settings for other servers
+  def self.update_other_server_configs(opts)
+    url = opts[:external_url]
+    url = MiqCockpit::WS.url(opts, MiqServer.my_server.ipaddress) unless url
+
+    MiqServer.active_miq_servers.where.not(:id => MiqServer.my_server.id) do |server|
+      conf = server.get_config
+      conf[:workers][:cockpit_ws_worker] = {
+        :external_url => url
+      }
+      server.set_config(conf)
+    end
```

Comment:
If the goal is to have a region level setting, then this should be set via MiqRegion.

One big downside to this is that as active_miq_servers changes, this configuration won't be updated for those services.